ha/core/controllers/pcs/node_controller.py

```diff
@@ -273,34 +275,35 @@ def stop(self, node_id: str, timeout: int= -1, **op_kwargs) -> dict:
         """
         Stop Node with nodeid.
         Args:
-            nodeid (str): Node ID from cluster nodes.
+            node_id (str): Node ID from cluster nodes.
         Returns:
             ([dict]): Return dictionary. {"status": "", "output": "", "error": ""}
                 status: Succeeded, Failed, InProgress
         """
+        # Get the node_name (pvtfqdn) fron nodeid
+        node_name = self._get_node_name(node_id=node_id)
         # TODO: check_cluster - boolean.It checks whether cluster is going to be offline if node with node_id is stopped.
         timeout = const.NODE_STOP_TIMEOUT if timeout < 0 else timeout
         # TODO: Get the node_status from system health status
-        node_status = self.nodes_status([node_id]).get(node_id)
+        node_status = self.nodes_status([node_name]).get(node_name)
         if node_status == NODE_STATUSES.CLUSTER_OFFLINE.value:
-            Log.info(f"For stop {node_id}, Node already in offline state.")
-            status = f"Node {node_id} is already in offline state."
+            Log.info(f"For stop {node_name}, Node already in offline state.")
+            status = f"Node {node_name} is already in offline state."
         elif node_status == NODE_STATUSES.POWEROFF.value:
-            raise ClusterManagerError(f"Failed to stop {node_id}. Node is in {node_status}.")
+            raise ClusterManagerError(f"Failed to stop {node_name}. Node is in {node_status}.")
         else:
-            if self.heal_resource(node_id):
+            if self.heal_resource(node_name):
                 time.sleep(const.BASE_WAIT_TIME)
             try:
-                Log.info(f"Please Wait, trying to stop node: {node_id}")
-                self._execute.run_cmd(const.PCS_STOP_NODE.replace("<node>", node_id)
-                        .replace("<seconds>", str(timeout)))
-                Log.info(f"Executed node stop for {node_id}, Waiting to stop resource")
+                Log.info(f"Please Wait, trying to stop node: {node_name}")
+                self._execute.run_cmd(f"pcs cluster stop {node_name}")
```

Comment:
Please add a TODO here to add timeout value and use PCS_STOP_NODE from const.py

Reply:
Added TODO